helper/resource/testing.go

```diff
@@ -1632,6 +2023,51 @@ func TestCheckOutput(name, value string) TestCheckFunc {
 	}
 }
 
+// Deprecated: State checks using [TestStep.Check] have been superseded by [TestStep.ConfigStateChecks].
+// Use the built-in [statecheck.ExpectKnownOutputValue], and
+// [statecheck.ExpectKnownOutputValueAtPath] state checks, can be used in
+// combination with a [custom] [knownvalue.Check] instead.
+// TestMatchOutput function will be removed in the next major version.
+//
+// The following is an example of using [statecheck.ExpectKnownOutputValueAtPath]
+// in combination with a [custom] [knownvalue.Check] to replicate the behaviour of
+// TestMatchOutput.
+//
+//	package example_test
+//
+//	import (
+//		"testing"
+//
+//		"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+//		"github.com/hashicorp/terraform-plugin-testing/statecheck"
+//		"github.com/hashicorp/terraform-plugin-testing/tfjsonpath"
+//	)
+//
+//	func TestExpectKnownOutputValueAtPath_CheckState_String_Custom(t *testing.T) {
+//		t.Parallel()
+//
+//		resource.Test(t, resource.TestCase{
+//			// Provider definition omitted.
+//			Steps: []resource.TestStep{
+//				{
+//					// Example resource containing a computed string attribute named "computed_attribute"
+//					Config: `resource "test_resource" "one" {}
+//
+//							output test_resource_one_output {
+//								value = test_resource.one
+//							}`,
```

Comment:
Similar here regarding an additional code comment to discourage the practice, if its resource-based.